frontend/src/components/taste/TasteList.vue

```diff
@@ -1,88 +1,80 @@
+                :class="{added: isChecked(taste.id), pop: isAnimated(taste.id), popin: !isClicked(taste.id)}"
                 :key="taste.id"
                 :style="{ animationDelay: index * 0.02 + 's' }"
             >
-                <div class="pill" @click="checkTaste(index)">
+                <div class="pill" @click="checkTaste(taste.id)">
                     {{ taste.name }}
-                    <i v-if="taste.check" class="fas fa-check" />
+                    <i v-if="isChecked(taste.id)" class="fas fa-check" />
                     <i v-else class="fas fa-plus" />
                 </div>
             </li>
         </ul>
-        <div class="loadMore">Load more</div>
-        <button class="doneButton loggedOut">Save and register</button>
     </div>
 </template>
 
 <script>
+import { mapState, mapGetters } from 'vuex';
+
 export default {
     name: 'TasteList',
     data() {
         return {
             timeDelay: 0,
-            tastes: [
-                {
-                    id: 1,
-                    name: 'Craft beer',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 2,
-                    name: 'Ice-cream',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 3,
-                    name: 'Live-music',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 4,
-                    name: 'Sport-bar',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 5,
-                    name: 'Lunch',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 6,
-                    name: 'Sushi',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                }
-            ]
+            selectedIds: [],
+            tastesData: []
         };
     },
     methods: {
-        checkTaste(index) {
-            let taste = this.tastes[index];
-            taste.check = !taste.check;
-            taste.isClick = true;
-            taste.isAnimate = true;
+        checkTaste(id) {
+            if (this.tastesData[id] === undefined) {
+                this.tastesData[id] = {
+                    check: false,
+                    isClick: false,
+                    isAnimate: false
+                };
+            }
+            let tasteData = this.tastesData[id];
+            if (this.selectedIds.includes(id)) {
+                tasteData.check = false;
```

Comment:
This could be another method
```
                tasteData.check = false;
                this.$store.dispatch('taste/deleteUserTaste', id);
                this.selectedIds.splice(this.selectedIds.indexOf(id), 1)
```